test/unit_test/test_general_functions.py

```diff
@@ -0,0 +1,133 @@
+                  "byzxnoxxbtofhnpdepatvbihjoungenjidckhepgdlsmnrbqdgaalidw" \
+                  "gccbardglcnedcqqxduuaauzyv"
+        str_500 = str_250 + str_250
+        more_than_500_char_even = \
+            str_250 + less_than_500_char + less_than_500_char + str_250
+        more_than_500_char_odd = \
+            str_250 + less_than_500_char + one_char + less_than_500_char + \
+            str_250
+        middle = '\u2026 ' + newline + newline + '\u2026'
+        assert make_preview_from(less_than_500_char) == less_than_500_char
+        assert make_preview_from(str_500) == str_500
+
+        assert make_preview_from(
+            more_than_500_char_odd) == str_250 + middle + str_250
+        assert make_preview_from(
+            more_than_500_char_even) == str_250 + middle + str_250
+
+    def test_generate_d3_object(self):
+        assert generate_d3_object(
+            {'a': 1, 'b': 2, 'c': 3, 'd': 4},
+            "object", "word", "count") == {
+            'name': 'object', 'children': [{'word': 'a', 'count': 1},
+                                           {'word': 'b', 'count': 2},
+                                           {'word': 'c', 'count': 3},
+                                           {'word': 'd', 'count': 4}]}
+
+    def test_merge_list(self):
+        assert merge_list([{"a": 1, "b": 2}, {"c": 3, "d": 4}]) == {
+            'a': 1, 'b': 2, 'c': 3, 'd': 4}
+
+    def test_load_stastic(self):
+        assert load_stastic(
+            "this is a string string") == {
+            "this": 1, "is": 1, "a": 1, "string": 2}
+
+    def test_matrix_to_dict(self):
+        assert matrix_to_dict([['', 'a', 'b', 'c', 'd'], [0, 1, 2, 3, 4]]) == \
+            [{'a': 1, 'b': 2, 'c': 3, 'd': 4}]
+
+    def test_dict_to_matrix(self):
+        assert dict_to_matrix(
+            [{'a': 1, 'b': 2, 'c': 3, 'd': 4}]) == (
+                [['', 'a', 'b', 'c', 'd'], [0, 1, 2, 3, 4]],
+                ['a', 'b', 'c', 'd'])
+
+
+class TestHtmlEscape:
+    def test_amp(self):
+        assert html_escape('&') == "&amp;"
+
+    def test_quot(self):
+        assert html_escape('"') == "&quot;"
+
+    def test_apos(self):
+        assert html_escape("'") == "&apos;"
+
+    def test_gt(self):
+        assert html_escape('>') == "&gt;"
+
+    def test_lt(self):
+        assert html_escape('<') == "&lt;"
+
+    def test_all(self):
+        assert html_escape('&"\'><') == '&amp;&quot;&apos;&gt;&lt;'
+        assert html_escape("<html lang='en'></html>") == '&lt;html lang=&apo' \
+                                                         's;en&apos;&gt;&lt;' \
+                                                         '/html&gt;'
+        assert html_escape('<html lang="en"></html>') == '&lt;html lang=&quo' \
+                                                         't;en&quot;&gt;&lt;' \
+                                                         '/html&gt;'
+
+
+class TestApplyFunctionExcludeTags:
+    input_str = "<tag>asdf</tag>"
```

Comment:
please create another test for longer string and other special cases:

- invalidly formatted xml
- empty string